mlrun/runtimes/pod.py

```diff
@@ -104,6 +104,46 @@ class KubeResourceSpec(FunctionSpec):
         "preemption_mode",
         "security_context",
     ]
+    _fields_to_strip = FunctionSpec._fields_to_strip + [
+        "volumes",
+        "volume_mounts",
+        "resources",
+        "replicas",
+        "image_pull_policy",
+        "service_account",
+        "image_pull_secret",
+        "node_name",
+        "node_selector",
+        "affinity",
+        "priority_class_name",
+        "tolerations",
+        "preemption_mode",
+        "security_context",
+    ]
+    _fields_to_exclude_for_k8s_serialization = [
+        "volumes",
+        "volume_mounts",
+        "resources",
+        "env",
+        "image_pull_policy",
+        "service_account",
+        "image_pull_secret",
+        "node_name",
+        "node_selector",
+        "affinity",
+        "tolerations",
+        "security_context",
+    ]
+    _fields_to_exclude_for_serialization = (
+        FunctionSpec._fields_to_exclude_for_serialization
+        + _fields_to_exclude_for_k8s_serialization
+    )
+    _fields_to_exclude_for_enrichment = (
+        FunctionSpec._fields_to_exclude_for_enrichment
+        + [
+            "env",
```

Comment:
Do we want to exclude all env variables? We just want to strip the `v3io` related env variables, don't we?

Reply:
I'll add more explanation about the `fields_to_exclude_for_enrichment` eventually if you look in the `_enrich_field` implementation of the `KubeResourceSpec` you can see that we are strip the v3io related variables but return the whole env struct, so we don't lose the other envs